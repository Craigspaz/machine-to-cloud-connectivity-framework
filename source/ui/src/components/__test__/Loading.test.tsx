// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { render } from '@testing-library/react';
import { LoadingProgressBar, LoadingSpinner } from '../Loading';

test('renders the loading bar component', async () => {
  expect(render(<LoadingProgressBar loading={true} />).container).toMatchSnapshot();
});

test('renders no loading bar component', async () => {
  expect(render(<LoadingProgressBar loading={false} />).container).toMatchSnapshot();
});

test('renders the loading spinner component', async () => {
  expect(render(<LoadingSpinner loading={true} />).container).toMatchSnapshot();
});

test('renders no loading spinner component', async () => {
  expect(render(<LoadingSpinner loading={false} />).container).toMatchSnapshot();
});
