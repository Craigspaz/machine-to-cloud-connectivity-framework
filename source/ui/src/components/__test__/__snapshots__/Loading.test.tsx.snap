// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders no loading bar component 1`] = `<div />`;

exports[`renders no loading spinner component 1`] = `<div />`;

exports[`renders the loading bar component 1`] = `
<div>
  <div
    class="progress"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="100"
      class="progress-bar progress-bar-animated progress-bar-striped"
      role="progressbar"
      style="width: 100%;"
    />
  </div>
</div>
`;

exports[`renders the loading spinner component 1`] = `
<div>
  <span
    aria-hidden="true"
    class="spinner-border spinner-border-sm"
    role="status"
  />
  <span>
    Â 
  </span>
</div>
`;
